lib/surface/components/form/text_input.ex

```diff
@@ -34,18 +36,38 @@ defmodule Surface.Components.Form.TextInput do
   @doc "Keyword with options to be passed down to `text_input/3`"
   property opts, :keyword, default: []
 
+  context get form, from: Form, as: :form_context
+
   def render(assigns) do
+    form = get_form(assigns)
+    props = get_non_nil_props(assigns, [:value, :class])
+
     ~H"""
-    {{
-      text_input(
-        String.to_atom(@form),
-        @field,
-        [
-          value: @value,
-          class: @class,
-        ] ++ @opts
-      )
-    }}
+      {{
+        text_input(
+          form,
+          String.to_atom(@field),
+          props ++ @opts
+        )
+      }}
```

Comment:
Let's keep the content aligned with the `~H"""`, otherwise it will add extra spaces:

```elixir
~H"""
{{
  text_input(
    form,
    String.to_atom(@field),
    props ++ @opts
  )
}}
"""
```